packages/image_picker/example/lib/main.dart

```diff
@@ -33,52 +34,454 @@ class MyHomePage extends StatefulWidget {
-        title: const Text('Image Picker Example'),
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
       ),
-      body: new Center(
-        child: new FutureBuilder<File>(
-          future: _imageFile,
-          builder: (BuildContext context, AsyncSnapshot<File> snapshot) {
-            if (snapshot.connectionState == ConnectionState.done &&
-                snapshot.data != null) {
-              return new Image.file(snapshot.data);
-            } else if (snapshot.error != null) {
-              return const Text('error picking image.');
-            } else {
-              return const Text('You have not yet picked an image.');
-            }
-          },
-        ),
+      body: Center(
+        child: isVideo ? _previewVideo(_controller) : _previewImage(),
       ),
-      floatingActionButton: new Column(
+      floatingActionButton: Column(
         mainAxisAlignment: MainAxisAlignment.end,
         children: <Widget>[
-          new FloatingActionButton(
-            onPressed: () => _onImageButtonPressed(ImageSource.gallery),
+          FloatingActionButton(
+            onPressed: () {
+              isVideo = false;
+              _onImageButtonPressed(ImageSource.gallery);
+            },
+            heroTag: 'image0',
             tooltip: 'Pick Image from gallery',
             child: const Icon(Icons.photo_library),
           ),
-          new Padding(
+          Padding(
             padding: const EdgeInsets.only(top: 16.0),
-            child: new FloatingActionButton(
-              onPressed: () => _onImageButtonPressed(ImageSource.camera),
+            child: FloatingActionButton(
+              onPressed: () {
+                isVideo = false;
+                _onImageButtonPressed(ImageSource.camera);
+              },
+              heroTag: 'image1',
               tooltip: 'Take a Photo',
               child: const Icon(Icons.camera_alt),
             ),
           ),
+          Padding(
+            padding: const EdgeInsets.only(top: 16.0),
+            child: FloatingActionButton(
+              backgroundColor: Colors.red,
+              onPressed: () {
+                isVideo = true;
+                _onImageButtonPressed(ImageSource.gallery);
+              },
+              heroTag: 'video0',
+              tooltip: 'Pick Video from gallery',
+              child: const Icon(Icons.video_library),
+            ),
+          ),
+          Padding(
+            padding: const EdgeInsets.only(top: 16.0),
+            child: FloatingActionButton(
+              backgroundColor: Colors.red,
+              onPressed: () {
+                isVideo = true;
+                _onImageButtonPressed(ImageSource.camera);
+              },
+              heroTag: 'video1',
+              tooltip: 'Take a Video',
+              child: const Icon(Icons.videocam),
+            ),
+          ),
         ],
       ),
     );
   }
 }
+
+class AspectRatioVideo extends StatefulWidget {
```

Comment:
Could we reduce code duplication from `video_player/example` to the bare minimum, please?

The `image_picker` example should be focused on the functionality of image/video picking, not video playback. I think having the `AspectRatioVideo` widget here is OK, but I don't think we need play/pause controls for this example. And the `buildCard` method further below is not used at all.

Reply:
Okay sounds good. View-Only of the video works too. I had brought that over from the video player example because when the video is looping it will play noise and the user may want to silence it. But it should be up to the developer to implement playback anyway. I can remove everything but the viewer.

Reply:
Thanks. The merge conflict happens because I just removed all `.gitignore` files except the repo root one.

Reply:
Ok I resolved the conflict. Vscode seems to generate a lot of extra files everytime the project is run. Ill see if Android Studio is doing the same thing

Reply:
I wonder why Vscode generates eclipse project files?

Reply:
Not sure. maybe an extension in Vscode doing it. Ill check it out.